pkg/tls/tls.go

```diff
@@ -286,6 +295,20 @@ func readConfig(secrets corev1controllers.SecretController, acmeDomains []string
 	return ca, noCACerts, opts, nil
 }
 
+func getClusterIP(services corev1controllers.ServiceController) (string, error) {
+	service, err := services.Get(namespace.System, "rancher", metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return "", nil
+		}
+		return "", err
+	}
+	if service.Spec.ClusterIP == "" {
```

Comment:
This will cause rancher to restart if service IP is not assigned. But it should be rare since service are always created and IP should be ready when rancher is started.